DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -85,15 +85,29 @@ public IActionResult Section(int customisationId, int sectionId)
         public IActionResult Tutorial(int customisationId, int sectionId, int tutorialId)
         {
             var candidateId = User.GetCandidateId();
-            sessionService.StartOrUpdateSession(candidateId, customisationId, HttpContext.Session);
+            var centreId = User.GetCentreId();
 
             var tutorialContent =
                 tutorialContentService.GetTutorialContent(candidateId, customisationId, sectionId, tutorialId);
 
-            if (tutorialContent == null)
+            if (tutorialContent == null || centreId == null)
             {
                 return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
             }
+
+            var progressId = courseContentService.GetOrCreateProgressId(candidateId, customisationId, centreId.Value);
+
+            if (progressId == null)
+            {
+                logger.LogError(
+                    "Redirecting to 500 as no progress id was returned. " +
+                    $"Candidate id: {candidateId}, customisation id: {customisationId}, centre id: {centreId}");
+                return RedirectToAction("StatusCode", "LearningSolutions", new { code = 500 });

```

Comment:
Actually I was thinking about this and I think it makes more sense to redirect to 404 here. Because this would happen if for example the user tried to access a course which isn't offered by their centre. Would you mind fixing that for the initial menu page too (either on this or your other PR)? 🙂

Reply:
Yep, done for `Tutorial` in 5c1c564 🙂 I'll do the rest in the other PR